assets/scripts/admin/questionEditor.js

```diff
@@ -632,76 +632,118 @@ $(document).on('ready pjax:scriptcomplete', function () {
+    hideLabelSetAlert();
     return $.ajax({
       url: languageJson.lsdetailurl,
       data: {sid, lid},
       cache: true,
       success(json /*: {results: Array<{label_name: string, labels: Array<{code: string, title: string}>}>, languages: {}} */) {
-        if (json.languages === []) {
-          alert('Internal error: No languages');
-          throw 'abort';
-        }
+        if (json.success !== true) {
+          $('#labelsetpreview').empty();
+          showLabelSetAlert(languageJson.labelSetNotFound, 'danger'); // This could mean the label set is not found or it has no languages
+        } else {
+          if (json.languages === []) {
+            alert('Internal error: No languages');
+            throw 'abort';
+          }
 
-        const $liTemplate = $('<li role="presentation"></li>');
-        const $aTemplate = $('<a data-toggle="tab"></a>');
-        const $tabTodyTemplate = $('<div></div>');
-        const $listTemplate = $('<div class="list-group selector_label-list"></div>');
-        const $listItemTemplate = $('<div class="list-group-item row selector_label-list-row"></div>');
-        const $tabindex = $('<ul class="nav nav-tabs" role="tablist"></ul>');
-        const $tabbody = $('<div class="tab-content" style="max-height: 50vh; overflow:auto;"></div>');
-
-        var hasInvalidCodes = false;
-        const source = $('#labelsetbrowserModal').data('source');
-        const i = 0;
-        $.each(json.languages, (language, languageName) => {
-          const $linkItem = $aTemplate.clone();
-          const $bodyItem = $tabTodyTemplate.clone();
-          let $itemList = $listTemplate.clone();
-
-          const classLink = i === 0 ? 'active' : '';
-          const classBody = i === 0 ? 'tab-pane tab-pane fade in active' : 'tab-page tab-pane fade';
-
-          $linkItem.addClass(classLink).attr('href', `#language_${language}`).text(languageName);
-          $liTemplate.clone().append($linkItem).appendTo($tabindex);
-
-          $bodyItem.addClass(classBody).attr('id', `language_${language}`);
-          $tabbody.append($bodyItem);
-
-          const labelSet = json.results[language];
-
-          $itemList = $listTemplate.clone();
-
-          labelSet.labels.forEach((label) => {
-            // Label title is not concatenated directly because it may have non-encoded HTML
-            const $labelTitleDiv = $('<div class="col-md-7"></div>');
-            $labelTitleDiv.text(label.title);
-            const $listItem = $listItemTemplate.clone();
-            $listItem.append(`<div class="col-md-5 text-right" style="border-right: 4px solid #cdcdcd">${label.code}</div>`);
-            $listItem.append($labelTitleDiv);
-            $listItem.attr('data-label', JSON.stringify(label));
-            $itemList.append($listItem);
-
-            if (source === 'answeroptions' && label.code.length > 5) {
-              hasInvalidCodes = true;
+          const $liTemplate = $('<li role="presentation"></li>');
```

Comment:
Good example when using the lightest version of React or Vue could actually help with code organization. :) But no need to change now.